lib/prot/to2/msg67.c

```diff
@@ -74,11 +74,22 @@ int32_t msg67(fdo_prot_t *ps)
 		goto err;
 	}
 
-	if (rec_maxDeviceServiceInfoSz > 0 &&
-		rec_maxDeviceServiceInfoSz < MAXDEVICESERVICEINFOSZ) {
-		ps->maxDeviceServiceInfoSz = rec_maxDeviceServiceInfoSz;
-	} else {
-		ps->maxDeviceServiceInfoSz = MAXDEVICESERVICEINFOSZ;
+	LOG(LOG_DEBUG, "TO2.OwnerServiceInfoReady: Received maxDeviceServiceInfoSz = %d\n",
+		rec_maxDeviceServiceInfoSz);
+	if (rec_maxDeviceServiceInfoSz <= MIN_SERVICEINFO_SZ) {
+		// default to minimum and log it
+		ps->maxDeviceServiceInfoSz = MIN_SERVICEINFO_SZ;
+		LOG(LOG_DEBUG,
+			"TO2.OwnerServiceInfoReady: Received maxDeviceServiceInfoSz is less than "
+			"the minimum size supported. Defaulting to %d\n",
+			ps->maxDeviceServiceInfoSz);
+	}
+	else if	(rec_maxDeviceServiceInfoSz >= ps->maxDeviceServiceInfoSz) {
+		// nothing to do, just log it
+		LOG(LOG_DEBUG,
+			"TO2.OwnerServiceInfoReady: Received maxDeviceServiceInfoSz is more than "
+			"the maximum size supported. Defaulting to %d\n",
+			ps->maxDeviceServiceInfoSz);
```

Comment:
There should be an else clause where you assign `ps->maxDeviceServiceInfoSz = rec_maxDeviceServiceInfoSz`